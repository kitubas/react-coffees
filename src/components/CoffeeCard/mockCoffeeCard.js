export const coffee = {
    title: 'title 1',
    description: 'body 1',
    image: 'img/img.png'
  }