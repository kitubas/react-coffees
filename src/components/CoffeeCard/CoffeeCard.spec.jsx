import { render, screen } from "@testing-library/react";
import { CoffeeCard } from ".";
import { coffee as coffeeMock }  from "./mockCoffeeCard";

describe("<CoffeeCard />", () => {
  it('should render CoffeeCard correctly', () => {
    render(<CoffeeCard coffee={coffeeMock} />);

    expect(screen.getByAltText(coffeeMock.title))
    .toHaveAttribute('src', 'img/img.png');

    expect(screen.getByRole('heading', { name: coffeeMock.title })).toBeInTheDocument();
    expect(screen.getByText(coffeeMock.description)).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const {container} = render(<CoffeeCard coffee={coffeeMock} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
