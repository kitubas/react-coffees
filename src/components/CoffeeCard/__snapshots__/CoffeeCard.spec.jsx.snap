// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CoffeeCard /> should match snapshot 1`] = `
<div
  class="coffee"
>
  <img
    alt="title 1"
    src="img/img.png"
  />
  <div
    class="coffee-content"
  >
    <h2>
      title 1
    </h2>
    <p>
      body 1
    </p>
  </div>
</div>
`;
